GalaxyZooTouchTable/GalaxyZooTouchTable/ViewModels/CenterpieceViewModel.cs

```diff
@@ -0,0 +1,87 @@
+﻿using GalaxyZooTouchTable.Lib;
+using GalaxyZooTouchTable.Models;
+using System.Collections.ObjectModel;
+using System.Collections.Specialized;
+using System.ComponentModel;
+using System.Windows.Threading;
+
+namespace GalaxyZooTouchTable.ViewModels
+{
+    public class CenterpieceViewModel : ViewModelBase
+    {
+        public ObservableCollection<TableUser> AllUsers { get; set; } = new ObservableCollection<TableUser>();
+
+        private bool _showJoinMessage = true;
+        public bool ShowJoinMessage
+        {
+            get => _showJoinMessage;
+            set => SetProperty(ref _showJoinMessage, value);
+        }
+
+        private bool _flipCenterpiece;
+        public bool FlipCenterpiece
+        {
+            get => _flipCenterpiece;
+            set => SetProperty(ref _flipCenterpiece, value);
+        }
+
+        public CenterpieceViewModel()
+        {
+            CreateTimer();
+
+            AllUsers.CollectionChanged += AllUsersCollectionChanged;
+
+            AllUsers.Add(CommonData.GetInstance().PersonUser);
+            AllUsers.Add(CommonData.GetInstance().LightUser);
+            AllUsers.Add(CommonData.GetInstance().StarUser);
+            AllUsers.Add(CommonData.GetInstance().HeartUser);
+            AllUsers.Add(CommonData.GetInstance().FaceUser);
+            AllUsers.Add(CommonData.GetInstance().EarthUser);
+        }
+
+        private void AllUsersCollectionChanged(object sender, NotifyCollectionChangedEventArgs changedEventArgs)
+        {
+
+            if (changedEventArgs.NewItems != null)
+            {
+                foreach (object item in changedEventArgs.NewItems)
+                {
+                    ((INotifyPropertyChanged)item).PropertyChanged += ItemPropertyChanged;
+                }
+            }
+            if (changedEventArgs.OldItems != null)
+            {
+                foreach (object item in changedEventArgs.OldItems)
+                {
+                    ((INotifyPropertyChanged)item).PropertyChanged -= ItemPropertyChanged;
+                }
+            }
+        }
+
+        private void ItemPropertyChanged(object sender, PropertyChangedEventArgs changedEventArgs)
+        {
+            foreach (TableUser user in AllUsers)
```

Comment:
You may want to use `find` here to save some lines https://docs.microsoft.com/en-us/dotnet/api/system.array.find?view=netframework-4.7.2.

Reply:
`find` isn't available on an ObservableCollection, but I can shorten these lines by using LINQ